framework/source/class/qx/ui/decoration/MBackgroundImage.js

```diff
@@ -90,48 +110,85 @@ qx.Mixin.define("qx.ui.decoration.MBackgroundImage",
      */
     _styleBackgroundImage : function(styles)
     {
-      var image = this.getBackgroundImage();
-      if(!image) {
-        return;
-      }
-
-      var id = qx.util.AliasManager.getInstance().resolve(image);
-      var source = qx.util.ResourceManager.getInstance().toUri(id);
-      if (styles["background-image"]) {
-        styles["background-image"] +=  ', url(' + source + ')';
+      // transform non-array values to an array containing that value
+      var images = this.getBackgroundImage();
+      if(!images) { return; }
+      if(!qx.lang.Type.isArray(images)) images = [images];
+      var repeats = this.getBackgroundRepeat();
+      if(!qx.lang.Type.isArray(repeats)) repeats = [repeats];
+      var tops = this.getBackgroundPositionY();
+      if(!qx.lang.Type.isArray(tops)) tops = [tops];
+      var lefts = this.getBackgroundPositionX();
+      if(!qx.lang.Type.isArray(lefts)) lefts = [lefts];
+      var origins = this.getBackgroundOrigin();
+      if(!qx.lang.Type.isArray(origins)) origins = [origins];
+
```

Comment:
If you want to make sure you get an array back ... you should have a function that takes care of this and not just keep copy pasting the code ... the only real difference here is the property you are looking at, so I would expect to see something where you have a list of properties you are interested in and some code that takes that list as input. DRY